src/Symfony/Component/Workflow/Debug/TraceableWorkflow.php

```diff
@@ -0,0 +1,120 @@
+ */
+
+namespace Symfony\Component\Workflow\Debug;
+
+use Symfony\Component\Stopwatch\Stopwatch;
+use Symfony\Component\Workflow\Definition;
+use Symfony\Component\Workflow\Marking;
+use Symfony\Component\Workflow\MarkingStore\MarkingStoreInterface;
+use Symfony\Component\Workflow\Metadata\MetadataStoreInterface;
+use Symfony\Component\Workflow\TransitionBlockerList;
+use Symfony\Component\Workflow\WorkflowInterface;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class TraceableWorkflow implements WorkflowInterface
+{
+    private array $calls = [];
+
+    public function __construct(
+        private readonly WorkflowInterface $workflow,
+        private readonly Stopwatch $stopwatch,
+    ) {
+    }
+
+    public function getMarking(object $subject, array $context = []): Marking
+    {
+        return $this->callInner(__FUNCTION__, \func_get_args());
+    }
+
+    public function can(object $subject, string $transitionName): bool
+    {
+        return $this->callInner(__FUNCTION__, \func_get_args());
+    }
+
+    public function buildTransitionBlockerList(object $subject, string $transitionName): TransitionBlockerList
+    {
+        return $this->callInner(__FUNCTION__, \func_get_args());
+    }
+
+    public function apply(object $subject, string $transitionName, array $context = []): Marking
+    {
+        return $this->callInner(__FUNCTION__, \func_get_args());
+    }
+
+    public function getEnabledTransitions(object $subject): array
+    {
+        return $this->callInner(__FUNCTION__, \func_get_args());
+    }
+
+    public function getName(): string
+    {
+        return $this->workflow->getName();
+    }
+
+    public function getDefinition(): Definition
+    {
+        return $this->workflow->getDefinition();
+    }
+
+    public function getMarkingStore(): MarkingStoreInterface
+    {
+        return $this->workflow->getMarkingStore();
+    }
+
+    public function getMetadataStore(): MetadataStoreInterface
+    {
+        return $this->workflow->getMetadataStore();
+    }
+
+    public function getCalls(): array
+    {
+        return $this->calls;
+    }
+
+    private function callInner(string $method, array $args): mixed
+    {
+        $sMethod = $this->workflow::class.'::'.$method;
+        $this->stopwatch->start($sMethod, 'workflow');
+
+        $previousMarking = null;
+        if ('apply' === $method) {
+            try {
+                $previousMarking = $this->workflow->getMarking($args[0]);
+            } catch (\Throwable) {
+            }
+        }
+
+        try {
+            return $return = $this->workflow->{$method}(...$args);
+        } catch (\Throwable $exception) {
+            throw $exception;
+        } finally {
+            $periods = $this->stopwatch->stop($sMethod)->getPeriods();
+            $period = end($periods);
+            $call = [
+                'method' => $method,
+                'duration' => $period->getDuration(),
```

Comment:
thx for having swap to stopwatch, what bout adding here:

```suggestion
                'memory' => $period->getMemory(),
```

would be awesome, as per internal usage we can plug some ±heavy listener/subscriber on workflow event, and having both duration and memory can help fast check on what cant be improved :)

Reply:
I'm not sure it's really useful, since here it's not per event, but for the whole method (`can`, `apply`, etc). So you won't know what use memory. 

I'm working on adding profiler for each listener - but to be honest, it's super hard :(

Reply:
copy that, lets see how it goes :)

Reply:
Profiling memort is also quite hard. As soon as you have nested periods, things become wonky because the memory used by the inner traceable object now counts as well for the outside span.
Memory profiling is better done with an actual profiler tool than this traceable instrumentation. It will be less skewed.

Reply:
>I'm working on adding profiler for each listener - but to be honest, it's super hard :(

@lyrixx It should be done by default if the workflow uses the event dispatcher service, it is traceable in debug and events should be listed in the Events profiler panel since always.

Reply:
@HeahDude indeed, the profiler shows some data, but it's not as good as what you can see in the Security profiler

### Workflow

![image](https://github.com/symfony/symfony/assets/408368/b184925d-9852-4f26-a3a8-9bbd4838a208)

### Security 

![image](https://github.com/symfony/symfony/assets/408368/366f1136-f5ac-4a12-8e92-ff5ba40a4747)


TL;DR:

I would like something with

* all listener called
  * timing information
* for the guard
  * the "response"